sources/lib/frontend/typechecker.ml

```diff
@@ -489,6 +490,20 @@ let rec type_term ?(call_from_type_form=false) env f =
           TTinfix(te1,s,te2) , ty1
         | _ -> error (ShouldHaveTypeInt ty1) t1.pp_loc
       end
+    | PPinfix(t1, PPpow, t2) ->
+      begin
+        let s = symbol_of PPpow in
+        let te1 = type_term env t1 in
+        let te2 = type_term env t2 in
+        let ty1 = Ty.shorten te1.c.tt_ty in
+        let ty2 = Ty.shorten te2.c.tt_ty in
+        match ty1, ty2 with
+        | Ty.Tint, Ty.Tint | Ty.Tint, Ty.Treal
+        | Ty.Treal, Ty.Tint | Ty.Treal, Ty.Treal  ->
+          Options.tool_req 1 (append_type "TR-Typing-OpMod type" ty1);
+          TTinfix(te1,s,te2) , ty1
+        | _ -> error (ShouldHaveTypeInt ty1) t1.pp_loc
```

Comment:
Unless I'm mistaken, this code says `t1` should have type `ty1` in order for the expression to be typecheck but that is already the case by definition, should there be a `t2` or `ty2` somewhere ?